### ISSUE TITLE
Cannot access to the Traefik Dashboard

### ISSUE THREAD

**Issue by thibaudbrg** [htpcBeginner/docker-traefik#254]
### Discussed in https://github.com/htpcBeginner/docker-traefik/discussions/253

<div type='discussions-op-text'>

<sup>Originally posted by **thibaudbrg** March  4, 2022</sup>
Hey! I still have a problem which is not the least... I can't access the dashboard...
I configured (by doing a near copy and paste of your docker-compose.yml available on GitHub) traefik2, watchtower, docker-gc and cf-companion. My sensitive data is with Docker Secrets. I configured my domain name with cloudflare, and my CNAMEs are automatically generated by cf-companion.
On my router, ports 80, 443 and 8080 are open and redirects to my Synology IP-Address. I've got a valid Let's Encrypt certificate in the acme.json file. Also thanks to a task, nginx does not uses ports 80 and 443 anymore so there are free to use.
In the logs, no errors, no warnings apppear...

However I can't access the dashboard. With: 
traefikh.$DOMAINENAME0/dashboard/    

Here are some pictures of my configuration if it can help.

PS: Why when my CNAMEs are created automatically, I have a home.$DOMAINENAME0? Where does the "home."
Thank you for your help!

![Screenshot 2022-03-04 094657](https://user-images.githubusercontent.com/73235385/156746629-0d18c343-e4bd-44d5-ab3c-a8d1327430d6.png)
![Screenshot 2022-03-04 095639](https://user-images.githubusercontent.com/73235385/156746672-df946b3f-a4d9-4490-ae3f-387eb73c7283.png)
![Screenshot 2022-03-04 095704](https://user-images.githubusercontent.com/73235385/156746690-4954a859-13c2-4c90-aa9a-f4d6c431299b.png)
![Screenshot 2022-03-04 105335](https://user-images.githubusercontent.com/73235385/156746700-43876660-97e3-4120-839c-10c8a800dc39.png)
![Screenshot 2022-03-04 105658](https://user-images.githubusercontent.com/73235385/156746702-feac5260-1b9f-46e2-a61d-52e49b6bbaa7.png)</div>

**Comment by g0Idfinger** [comment]
What is the $DONAINNAME variable set to in .env?

**Comment by thibaudbrg** [comment]
$DOMAINNAME0 is just my domain name domain.com
Actually, i don't really know why there is a "home." prefix automatically add on this configuration. This is cf-companion that automatically created them.
If I manually remove the "home." in the content DNS table on cloudflare, it still doesn't work...
And the screenshot error (ERROR 1016) appears without the use of "home."

**Comment by g0Idfinger** [comment]
Can you try to disable cf-companion and manually create your traefik entry in cloudflare,  delete the acme.json file(recreate, touch and chmod) and restart docker containers?  Your best bet is to only have Traefik and socket-proxy as the first containers to bring up and then once that is working bring on the others.  

You may also need to figure out where it is getting the `home` subdomain from.

**Comment by htpcBeginner** [comment]
To me this one appears to a simple DNS record missing error. All your services are pointed to home.domain.com. But you do not have a home A record or CNAME pointing to your home IP. You have 2 options:

1. Create a home A record and point it to your IP or create a home CNAME and point it to @ 
2. Change all your services to point to the root domain and not home - this means edit cf-companion as commented in the docker compose.

**Comment by g0Idfinger** [comment]
> 1. Create a home A record and point it to your IP or create a home CNAME and point it to @
If you have the free version of CF, this wont work, I tried to make this work yesterday and as soon as I got rid of the subdomain everything worked.

**Comment by thibaudbrg** [comment]
I've just followed the second option you made (I have a free clouflare account). and I managed to remove the home. prefix by modifying the label section in cf-companion.

I've just deleted the acme.json file to recreate a Let's Encrypt certificate (everything work on this side).

Also, I've had portainer in the docker-compose to check if the problem comes only from traefik.
So now, my CNAMEs look like this: 
![Screenshot 2022-03-04 193041](https://user-images.githubusercontent.com/73235385/156822833-af6bdde8-74fd-4a90-817e-d7c10f799246.png)


My port 9000 is open on my router and redirects to my synology. 
When I try to reach portainer, with: `portainer.MYROOTDOMAINNAME:9000` it stilll doesn't work. And something weird.. (sorry everything's very new for me), If I just enter: `portainer.MYROOTDOMAINNAME` (without the port), I access to my router web configuration page.

**Comment by g0Idfinger** [comment]
You will want to open ports 443 and 80 and thats it for your internal server IP address.  Traefik will take care of the rest. You should NOT be able to hit your router web config page from the outside internet,  this is a huge security risk.

**Comment by htpcBeginner** [comment]
As said above only 443 and 80 to traefik host. Nothing else. Do not forward 9000 to portainer.

cloudflare only allows certain ports. 9000 is not in it https://developers.cloudflare.com/fundamentals/get-started/network-ports/

**Comment by htpcBeginner** [comment]
And the fact that you have cf free has nothing to do with this. I have cf free

**Comment by g0Idfinger** [comment]
> And the fact that you have cf free has nothing to do with this. I have cf free

It was in response to the sub domain, which to my knowledge is will not work on free account, at least I couldn't get it to work.

**Comment by thibaudbrg** [comment]
Okay, I deleted de 9000 port forwarding, only 443 and 80 are forward.
Now I'm trying to access either traefik dashboard or portainer behind traefik, and I get the cf-error 522
![Screenshot 2022-03-04 205335](https://user-images.githubusercontent.com/73235385/156832618-8f5f50de-2897-45bf-b173-17c7bd2aaf3e.png)
Does this mean that traefik is bad configured?

**Comment by htpcBeginner** [comment]
Comment out api.insecure=true and try

**Comment by htpcBeginner** [comment]
Also check traefik logs as explained in the guides

**Comment by thibaudbrg** [comment]
The logs are ok,
It creates middleswares, and the server correctly. I've commented out api.insecure=true

Still same error.

My DDNS is configured in the synology DDNS page and it's work correctly, but my router is not configured with the DDNS. Do I have to do it? I'm kind of confused because in the DDNS section in my router, I cannot add a DDNS provided by Cloudflare and it has no way around to configure it manually

**Comment by g0Idfinger** [comment]
As long as you have ddns setup some where you should be good. Go to Ipchicken.com and make sure that ip addresses matches you CF records for your domain. If it is still not working, it is either you firewall(router or server or both) or something not right in the docker config.

**Comment by thibaudbrg** [comment]
I managed to access to traefik in local through `localhost:8080` and then I followed the instructions to quickly configure Traefik Pilot (by adding the `-- pilot.token=` line)...
But now I can't access the dashboard. I get `404 page not found`.

So I deleted my Traefik Pilot instance, deleted the line `--pilot.token=` but nothing changes (I removed the containers, deleted the images and started again from scratch). I can't even access the dashboard locally anymore. Is there something I missed when deleting the Pilot config?

In the logs, everything's ok, my traefik server is created and the middlewares to.
